base/broadcast.jl

```diff
@@ -919,20 +919,19 @@ end
     length(dest) < 256 && return invoke(copyto!, Tuple{AbstractArray, Broadcasted{Nothing}}, dest, bc)
```

Comment:
Perhaps this should be lowered now if the new implementation is faster.

Reply:
Yes, I can explore the small array performance a little more closely to tune this.  I was focusing on the overall performance of `partition` here.

Reply:
Turns out it doesn't really change anything because almost all the time is spent allocating/zeroing the `tmp` vector — there's not much room for a faster algorithm to make up.  I commented this line out and used your `opt` benchmark from #33517:

```julia
julia> opt(256);
  370.055 ns (1 allocation: 4.19 KiB)
  371.330 ns (0 allocations: 0 bytes)

julia> f() = (x = Vector{Bool}(undef, Base.bitcache_size); @simd for i in eachindex(x) @inbounds x[i] = 0; end; x)
f (generic function with 1 method)

julia> @btime f();
  311.239 ns (1 allocation: 4.19 KiB)
```

I think we can keep this the same.

Reply:
Cool, thanks for trying it out.